Teraform plan fails with azurerm and site_config in function_app resource v.0.14.0-beta1

### Terraform Version
```
Terraform v0.14.0-beta1
+ provider registry.terraform.io/hashicorp/azurerm v2.31.0
```

### Terraform Configuration Files

```terraform
variable "sensitiveString" {
  type = string
  default = "thisIsNothing"
  sensitive = true
}

resource "azurerm_function_app" "afapp" {
  name                      = "someName"
  location                  = azurerm_resource_group.rg.location
  resource_group_name       = azurerm_resource_group.rg.name
  app_service_plan_id       = azurerm_app_service_plan.appsp.id
  storage_connection_string = azurerm_storage_account.stacc.primary_connection_string
  version                   = "~3"

  identity {
    type = "SystemAssigned"
  }

  app_settings = {
    "_ReleaseName" = ""
    "_BuildName"   = ""
    "_BuildID"     = ""

    "stringFromKeyvault" = azurerm_key_vault_secret.keyvault_var.value

    "newStringSecret" = var.sensitiveString # ---> If this is here, it crashes the execution
  }

  lifecycle {
    ignore_changes = [
      app_settings["_BuildID"],
      app_settings["_BuildName"],
      app_settings["_ReleaseName"]
    ]
  }
}
```
### Crash Output
```
panic: value is marked, so must be unmarked first

goroutine 364 [running]:
github.com/zclconf/go-cty/cty.Value.assertUnmarked(...)
	/go/pkg/mod/github.com/zclconf/go-cty@v1.6.2-0.20201013200640-e5225636c8c2/cty/marks.go:123
github.com/zclconf/go-cty/cty.Value.True(0x2f283c0, 0xc000050a71, 0x284b120, 0xc0009c79e0, 0xc00142f560)
	/go/pkg/mod/github.com/zclconf/go-cty@v1.6.2-0.20201013200640-e5225636c8c2/cty/value_ops.go:360 +0x59
github.com/zclconf/go-cty/cty.Value.False(0x2f283c0, 0xc000050a71, 0x284b120, 0xc0009c79e0, 0x2f28301)
	/go/pkg/mod/github.com/zclconf/go-cty@v1.6.2-0.20201013200640-e5225636c8c2/cty/value_ops.go:369 +0x50
github.com/zclconf/go-cty/cty.Value.Equals(0x2f28480, 0xc001644b60, 0x275a740, 0xc001294c60, 0x2f28480, 0xc000e08340, 0x275a740, 0xc00142f050, 0xc000e08340, 0x275a740, ...)
	/go/pkg/mod/github.com/zclconf/go-cty@v1.6.2-0.20201013200640-e5225636c8c2/cty/value_ops.go:315 +0x1686
github.com/hashicorp/terraform/terraform.processIgnoreChangesIndividual(0x2f284c0, 0xc0016454d0, 0x275a740, 0xc001295560, 0x2f284c0, 0xc000e08380, 0x275a740, 0xc00142f110, 0xc0007c02a0, 0x3, ...)
	/home/circleci/project/project/terraform/eval_diff.go:650 +0x635
github.com/hashicorp/terraform/terraform.(*EvalDiff).processIgnoreChanges(0xc0009d1a98, 0x2f284c0, 0xc0016454d0, 0x275a740, 0xc001295560, 0x2f284c0, 0xc000e08380, 0x275a740, 0xc00142f110, 0x0, ...)
	/home/circleci/project/project/terraform/eval_diff.go:575 +0x249
github.com/hashicorp/terraform/terraform.(*EvalDiff).Eval(0xc0009d1a98, 0x2f62560, 0xc000dce340, 0x0, 0x0, 0x0, 0x0)
	/home/circleci/project/project/terraform/eval_diff.go:210 +0x669
github.com/hashicorp/terraform/terraform.(*NodePlannableResourceInstance).managedResourceExecute(0xc0016bb8a0, 0x2f62560, 0xc000dce340, 0x0, 0xc00083c000, 0x8a)
	/home/circleci/project/project/terraform/node_resource_plan_instance.go:207 +0x5b3
github.com/hashicorp/terraform/terraform.(*NodePlannableResourceInstance).Execute(0xc0016bb8a0, 0x2f62560, 0xc000dce340, 0xc000032002, 0x27a77a0, 0x2aa4140)
	/home/circleci/project/project/terraform/node_resource_plan_instance.go:39 +0xf9
github.com/hashicorp/terraform/terraform.(*ContextGraphWalker).Execute(0xc000726c30, 0x2f62560, 0xc000dce340, 0x1c7f3bbec20, 0xc0016bb8a0, 0x0, 0x0, 0x0)
	/home/circleci/project/project/terraform/graph_walk_context.go:127 +0xc3
github.com/hashicorp/terraform/terraform.(*Graph).walk.func1(0x2aa4140, 0xc0016bb8a0, 0x0, 0x0, 0x0)
	/home/circleci/project/project/terraform/graph.go:59 +0x97e
github.com/hashicorp/terraform/dag.(*Walker).walkVertex(0xc000306ba0, 0x2aa4140, 0xc0016bb8a0, 0xc00065c980)
	/home/circleci/project/project/dag/walk.go:387 +0x375
created by github.com/hashicorp/terraform/dag.(*Walker).Update
	/home/circleci/project/project/dag/walk.go:309 +0x1246
```

### Expected Behavior
sensitive value should've been added to function app

### Actual Behavior
terraform crashed

### Steps to Reproduce

1. `terraform init`
2. `terraform apply`

--- a/terraform/context_plan_test.go
+++ b/terraform/context_plan_test.go
@@ -4667,6 +4667,65 @@ func TestContext2Plan_ignoreChangesInMap(t *testing.T) {
 	}), ric.After)
 }
 
+func TestContext2Plan_ignoreChangesSensitive(t *testing.T) {
+	m := testModule(t, "plan-ignore-changes-sensitive")
+	p := testProvider("aws")
+	p.PlanResourceChangeFn = testDiffFn
+
+	state := states.NewState()
+	root := state.EnsureModule(addrs.RootModuleInstance)
+	root.SetResourceInstanceCurrent(
+		mustResourceInstanceAddr("aws_instance.foo").Resource,
+		&states.ResourceInstanceObjectSrc{
+			Status:    states.ObjectReady,
+			AttrsJSON: []byte(`{"id":"bar","ami":"ami-abcd1234","type":"aws_instance"}`),
+		},
+		mustProviderConfig(`provider["registry.terraform.io/hashicorp/aws"]`),
+	)
+
+	ctx := testContext2(t, &ContextOpts{
+		Config: m,
+		Providers: map[addrs.Provider]providers.Factory{
+			addrs.NewDefaultProvider("aws"): testProviderFuncFixed(p),
+		},
+		Variables: InputValues{
+			"foo": &InputValue{
+				Value:      cty.StringVal("ami-1234abcd"),
+				SourceType: ValueFromCaller,
+			},
+		},
+		State: state,
+	})
+
+	plan, diags := ctx.Plan()
+	if diags.HasErrors() {
+		t.Fatalf("unexpected errors: %s", diags.Err())
+	}
+
+	schema := p.GetSchemaReturn.ResourceTypes["aws_instance"]
+	ty := schema.ImpliedType()
+
+	if len(plan.Changes.Resources) != 1 {
+		t.Fatal("expected 1 changes, got", len(plan.Changes.Resources))
+	}
+
+	res := plan.Changes.Resources[0]
+	ric, err := res.Decode(ty)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if ric.Addr.String() != "aws_instance.foo" {
+		t.Fatalf("unexpected resource: %s", ric.Addr)
+	}
+
+	checkVals(t, objectVal(t, schema, map[string]cty.Value{
+		"id":   cty.StringVal("bar"),
+		"ami":  cty.StringVal("ami-abcd1234"),
+		"type": cty.StringVal("aws_instance"),
+	}), ric.After)
+}
+
 func TestContext2Plan_moduleMapLiteral(t *testing.T) {
 	m := testModule(t, "plan-module-map-literal")
 	p := testProvider("aws")

--- a/terraform/eval_diff.go
+++ b/terraform/eval_diff.go
@@ -202,24 +202,24 @@ func (n *EvalDiff) Eval(ctx EvalContext) (interface{}, error) {
 		priorVal = cty.NullVal(schema.ImpliedType())
 	}
 
+	// Create an unmarked version of our config val and our prior val.
+	// Store the paths for the config val to re-markafter
+	// we've sent things over the wire.
+	unmarkedConfigVal, unmarkedPaths := origConfigVal.UnmarkDeepWithPaths()
+	unmarkedPriorVal, priorPaths := priorVal.UnmarkDeepWithPaths()
+
 	// ignore_changes is meant to only apply to the configuration, so it must
 	// be applied before we generate a plan. This ensures the config used for
 	// the proposed value, the proposed value itself, and the config presented
 	// to the provider in the PlanResourceChange request all agree on the
 	// starting values.
-	configValIgnored, ignoreChangeDiags := n.processIgnoreChanges(priorVal, origConfigVal)
+	configValIgnored, ignoreChangeDiags := n.processIgnoreChanges(unmarkedPriorVal, unmarkedConfigVal)
 	diags = diags.Append(ignoreChangeDiags)
 	if ignoreChangeDiags.HasErrors() {
 		return nil, diags.Err()
 	}
 
-	// Create an unmarked version of our config val and our prior val.
-	// Store the paths for the config val to re-markafter
-	// we've sent things over the wire.
-	unmarkedConfigVal, unmarkedPaths := configValIgnored.UnmarkDeepWithPaths()
-	unmarkedPriorVal, priorPaths := priorVal.UnmarkDeepWithPaths()
-
-	proposedNewVal := objchange.ProposedNewObject(schema, unmarkedPriorVal, unmarkedConfigVal)
+	proposedNewVal := objchange.ProposedNewObject(schema, unmarkedPriorVal, configValIgnored)
 
 	// Call pre-diff hook
 	if !n.Stub {
@@ -238,7 +238,7 @@ func (n *EvalDiff) Eval(ctx EvalContext) (interface{}, error) {
 	validateResp := provider.ValidateResourceTypeConfig(
 		providers.ValidateResourceTypeConfigRequest{
 			TypeName: n.Addr.Resource.Type,
-			Config:   unmarkedConfigVal,
+			Config:   configValIgnored,
 		},
 	)
 	if validateResp.Diagnostics.HasErrors() {
@@ -247,7 +247,7 @@ func (n *EvalDiff) Eval(ctx EvalContext) (interface{}, error) {
 
 	resp := provider.PlanResourceChange(providers.PlanResourceChangeRequest{
 		TypeName:         n.Addr.Resource.Type,
-		Config:           unmarkedConfigVal,
+		Config:           configValIgnored,
 		PriorState:       unmarkedPriorVal,
 		ProposedNewState: proposedNewVal,
 		PriorPrivate:     priorPrivate,
@@ -286,7 +286,7 @@ func (n *EvalDiff) Eval(ctx EvalContext) (interface{}, error) {
 		return nil, diags.Err()
 	}
 
-	if errs := objchange.AssertPlanValid(schema, unmarkedPriorVal, unmarkedConfigVal, plannedNewVal); len(errs) > 0 {
+	if errs := objchange.AssertPlanValid(schema, unmarkedPriorVal, configValIgnored, plannedNewVal); len(errs) > 0 {
 		if resp.LegacyTypeSystem {
 			// The shimming of the old type system in the legacy SDK is not precise
 			// enough to pass this consistency check, so we'll give it a pass here,

--- a/terraform/testdata/plan-ignore-changes-sensitive/ignore-changes-sensitive.tf
+++ b/terraform/testdata/plan-ignore-changes-sensitive/ignore-changes-sensitive.tf
@@ -1,0 +1,11 @@
+variable "foo" {
+  sensitive = true
+}
+
+resource "aws_instance" "foo" {
+  ami = var.foo
+
+  lifecycle {
+    ignore_changes = [ami]
+  }
+}

